docs/cow-protocol/concepts/order-types/pay-debt-flashloans.md

```diff
@@ -0,0 +1,89 @@
+---
+sidebar_position: 6
+---
+
+# Repay debt with collateral using flashloans
+
+A key advantage of flashloans is the ability to repay debt with collateral, since flashloans allow users to close or reduce their debt positions without needing upfront liquidity.
+
+Instead of requiring users to have assets readily available to repay their loans, flashloans enable them to temporarily borrow the needed funds, use those funds to repay their debt, and immediately reclaim their locked collateral. Once the collateral is received, it can be swapped or liquidated to cover the borrowed amount, ensuring the loan is repaid within the same transaction.
+
+This approach eliminates the need for users to preemptively sell assets or find external liquidity, making it a highly flexible and efficient way to manage debt positions.
+
+### How to do it
+
+The user can sign a pre-hook that deploys a [cowshed](../../reference/sdks/cow-sdk/classes/CowShedHooks.md) which pays back the debt using the flashloaned tokens. The underlying user order then it is just for paying back the required flashloan.
+
+This can be achieved through a buy order where the **buy token** is the flash-loaned asset, the **sell token** is the asset used as collateral, and the **sell amount** equals the full collateral amount. The receiver must always be the settlement contract, while the protocol ensures that the funds are sent to the appropriate address.
```

Comment:
More important than the `sellAmount` is the `buyAmount`. When building a `buy` order the `buyAmount` dictates how much will be bought and the `sellAmount` dictates the limit price. Since the transaction will revert if not enough is repaid it's more important to stress that the full flashloan amount needs to be bought and sent to the settlement contract.